src/app/EventManagement.cpp

```diff
@@ -583,31 +613,46 @@ CHIP_ERROR EventManagement::CopyEvent(const TLVReader & aReader, TLVWriter & aWr
 {
     TLVReader reader;
     TLVType containerType;
+    TLVType containerType1;
+    TLVType containerType2;
     CopyAndAdjustDeltaTimeContext context(&aWriter, apContext);
     CHIP_ERROR err = CHIP_NO_ERROR;
 
     reader.Init(aReader);
-    err = reader.EnterContainer(containerType);
-    SuccessOrExit(err);
-
-    err = aWriter.StartContainer(AnonymousTag, kTLVType_Structure, containerType);
-    SuccessOrExit(err);
+    ReturnErrorOnFailure(reader.EnterContainer(containerType));
+    ReturnErrorOnFailure(aWriter.StartContainer(AnonymousTag, kTLVType_Structure, containerType));
+
+    ReturnErrorOnFailure(reader.Next());
+    ReturnErrorOnFailure(reader.EnterContainer(containerType1));
+    ReturnErrorOnFailure(aWriter.StartContainer(TLV::ContextTag(to_underlying(EventReportIB::Tag::kEventStatus)),
+                                                kTLVType_Structure, containerType1));
+    ReturnErrorOnFailure(reader.Next());
+    do
+    {
+        ReturnErrorOnFailure(aWriter.CopyElement(reader));
+    } while (CHIP_NO_ERROR == (err = reader.Next()));
+    if (err == CHIP_END_OF_TLV)
+    {
+        err = CHIP_NO_ERROR;
+    }
+    ReturnErrorOnFailure(err);
+    ReturnErrorOnFailure(reader.ExitContainer(containerType1));
+    ReturnErrorOnFailure(aWriter.EndContainer(containerType1));
 
+    ReturnErrorOnFailure(reader.Next());
+    ReturnErrorOnFailure(reader.EnterContainer(containerType2));
+    ReturnErrorOnFailure(
+        aWriter.StartContainer(TLV::ContextTag(to_underlying(EventReportIB::Tag::kEventData)), kTLVType_Structure, containerType2));
     err = TLV::Utilities::Iterate(reader, CopyAndAdjustDeltaTime, &context, false /*recurse*/);
     if (err == CHIP_END_OF_TLV)
     {
         err = CHIP_NO_ERROR;
     }
-    SuccessOrExit(err);
-
-    err = aWriter.EndContainer(containerType);
-    SuccessOrExit(err);
-
-    err = aWriter.Finalize();
-    SuccessOrExit(err);
-
-exit:
-    return err;
+    ReturnErrorOnFailure(err);
+    ReturnErrorOnFailure(aWriter.EndContainer(containerType2));
```

Comment:
where aWriter StartContainer for containerType2

Reply:
aWriter.StartContainer(TLV::ContextTag(to_underlying(EventReportIB::Tag::kEventData)), kTLVType_Structure, containerType2)